### PR TITLE
Release 4.1-b4

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by zerothi** [comment]
Hi, thanks for trying this, I am still wondering why this suddenly stopped working! If you need any input, please do not hesitate to contact me!

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by jan-janssen** [comment]
@zerothi the tests are passing now, I excluded `TBTrans_rep` and `TBTrans`, because they seem to be no longer included with the new version. In addition I disabled the command part of the tests, maybe you can tell me which commands are included in the two packages above so I can remove only those. Finally I was not able to get the `nompi` version to work.

**Comment by zerothi** [comment]
Thanks!!! :)

Actually, I never did 4.1-b4 since I didn't want to have the default version being a beta release... :(
If this gets added, would installations not use this as the default?


As for tbtrans*. It has moved to Util/TS/TBtrans. But I think I can manage that, the main problem was the other thing.

File: .ci_support/linux_mpimpich.yaml [hunk 1]
```diff
@@ -14,12 +12,22 @@ fortran_compiler:
 - gfortran
 fortran_compiler_version:
 - '7'
+hdf5:
+- 1.10.5
+libblas:
+- 3.8 *netlib
```
Comment (zerothi) [inline .ci_support/linux_mpimpich.yaml:18]: Hmm, won't this rely on netlib blas?

File: .ci_support/linux_mpiopenmpi.yaml [hunk 2]
```diff
@@ -14,12 +12,22 @@ fortran_compiler:
 - gfortran
 fortran_compiler_version:
 - '7'
+hdf5:
+- 1.10.5
+libblas:
+- 3.8 *netlib
```
Comment (zerothi) [inline .ci_support/linux_mpiopenmpi.yaml:18]: also here?

File: .ci_support/osx_mpimpich.yaml [hunk 3]
```diff
@@ -1,29 +1,37 @@
...
 - conda-forge main
 fortran_compiler:
 - gfortran
 fortran_compiler_version:
-- '4'
+- '7'
+hdf5:
+- 1.10.5
+libblas:
+- 3.8 *netlib
```
Comment (zerothi) [inline .ci_support/osx_mpimpich.yaml:18]: and here

File: recipe/build.sh [hunk 4]
```diff
@@ -56,9 +56,10 @@ function mkcp {
 # First make a few of the libraries to check that they work!
 make libxmlparser.a
 # Try and build FoX to catch any debugs
-make FoX/.config || cat FoX/config.log
+# make FoX/.config || cat FoX/config.log
 
 mkcp siesta
+make clean-transiesta
 mkcp transiesta
```
Comment (zerothi) [inline recipe/build.sh:63]: In 4.1 you don't need this. I will fix this.

**Comment by jan-janssen** [comment]
I tried to use the same changes to build 4.0 but was not able to https://github.com/conda-forge/siesta-feedstock/pull/15

File: .ci_support/linux_mpimpich.yaml (hunk 1 above)
Comment (jan-janssen) [reply .ci_support/linux_mpimpich.yaml:18]: If openblas or mkl are available they are used instead. I was basically just following https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=blas#blas

Comment (zerothi) [reply .ci_support/linux_mpimpich.yaml:18]: Hmm, the way I read the documentation you should just have `libblas` no version specification, nor `*netlib`? It has changed some times during the anaconda period. I have tried to adapt, and now it seems to have changed again ;)

Hmm. I can try this when I get the time.

Comment (jan-janssen) [reply .ci_support/linux_mpimpich.yaml:18]: I use the same for quantum espresso and there it works fine https://github.com/conda-forge/qe-feedstock/blob/master/.ci_support/linux_.yaml

Comment (zerothi) [reply .ci_support/linux_mpimpich.yaml:18]: Yes, it works, but is it netlib BLAS?

File: recipe/build.sh (hunk 4 above)
Comment (jan-janssen) [reply recipe/build.sh:63]: Then I get the following error message: 
```
RUNNING: make transiesta
** You were building siesta in this directory.
** Please type 'make clean-transiesta' first 
make: *** [check-transiesta] Error 1
Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 474, in main
    execute(sys.argv[1:])
  File "/opt/conda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 465, in execute
    verify=args.verify, variants=args.variants)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 2864, in build_tree
    notest=notest,
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 1995, in build
    cwd=src_dir, stats=build_stats)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/utils.py", line 404, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/utils.py", line 384, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-o', 'err
```

Comment (zerothi) [reply recipe/build.sh:63]: Sorry, I wasn't clear. Transiesta command is not necessary. :)
So `make clean-transiesta`, `make transiesta` is not necessary. They can be emulated using a simple shell script
```bash
#!/bin/sh
siesta --electrode $@
```

File: .ci_support/linux_mpimpich.yaml (hunk 1 above)
Comment (jan-janssen) [reply .ci_support/linux_mpimpich.yaml:18]: No, here is the quantum espresso example: 
https://mybinder.org/v2/gh/jan-janssen/quantum-espresso-example/master
By default openblas is installed. The netlib part is only used during the compilation to make sure no openblas specific commands are used and the executable can run with mkl, openblas and netlib.

Comment (jan-janssen) [reply .ci_support/linux_mpimpich.yaml:18]:
```

libblas                   3.8.0               14_openblas    conda-forge
libcblas                  3.8.0               14_openblas    conda-forge
...
qe                        6.5                  h2fbfb5d_0    conda-forge
```

Comment (zerothi) [reply .ci_support/linux_mpimpich.yaml:18]: Ok, thanks for the clarification! :)

**Comment by jan-janssen** [comment]
> Actually, I never did 4.1-b4 since I didn't want to have the default version being a beta release... :(
> If this gets added, would installations not use this as the default?
Yes you are right, I made the mistake to name it `4.1.4` which is based on the requirement of conda to use numeric version names. We could use 4.1.0 and then just increase the build number once the official version is released. Still it would mean that conda users would switch to the beta version.

**Comment by jan-janssen** [comment]
closes https://github.com/conda-forge/siesta-feedstock/issues/9

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by jan-janssen** [comment]
I once more tried to build it with `nompi` but that does not work for some reason. So from my side the pull request is no ready to merge, but it is up to you to decide if you want to push a beta release to conda.

**Comment by jan-janssen** [comment]
I also added myself as a maintainer, then I can help with future releases.

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by zerothi** [comment]
Thanks, this is very welcome! :)
Let me do some changes. :)

**Pull request closed without merging**